### PR TITLE
CT-2975 Show third party name in case list

### PR DESCRIPTION
**Pull request by njpreddy**
## Description
CT-2975 Third party name in case list

## Self-review checklist

* [X] (1) Quick stakeholder demo done OR
* [ ] (2) ...bug with before and after screenshots
* [X] (3) Tests passing
* [X] (4) Branch ready to be merged (not work in progress)
* [X] (5) No superfluous changes in diff
* [X] (6) No TODO's without new ticket numbers
* [X] (7) PR Prefixed with ticket number e.g. `CT-7654 ...`

### Screenshots
State 1 - Non third party case displayed:

1  requester-requested

State 2 - Third party case with company name displayed:

2  third-party-company-name

State 3 - Third party case with contact name (no company name entered) displayed

3  third-party-contact-name

### Related JIRA tickets

https://dsdmoj.atlassian.net/secure/RapidBoard.jspa?rapidView=25&projectKey=CT&modal=detail&selectedIssue=CT-2975

### Manual testing instructions

State 1:
Create a case having "Requested on someone else's behalf?" set to 'No' and "where should it be sent?" set to either 'Data subject' or 'Third party'.  The cases list page will display the case requester's name duplicated in the "Request" column

State 2:
Create a case having "Requested on someone else's behalf?" set to 'Yes'.  Enter all required details, including a representative and company name.  The SAR can be sent to either data Subject or third party.  The cases list page will display the case requester's name with the third party company name below in the "Request" column

State3:
Edit the case created above byy deleting the company name.  The cases list page will now display the case requester's name with the third party contact name below in the "Request" column.

### CODE REVIEW

**Review by Polsonby — APPROVED** [review]
Great :) 

Ideally we'd have a view spec covering the logic here, but as long as the existing tests pass I'm happy to approve for now. Could we be sure to add a ticket to circle back to the view spec after migration?

File: app/helpers/cases_helper.rb [hunk 1]
```diff
@@ -274,8 +274,14 @@ def action_link_for_destroy_case_link(kase, linked_case)
 
   def request_details_html(kase)
     content_tag(:strong, "#{kase.subject} ", class: 'strong') +
-      if kase.type == "Case::SAR::Offender" && kase.third_party_name.present?
-        content_tag(:div, kase.third_party_name, class: 'case-name-detail')
+      if kase.type == "Case::SAR::Offender"
```
Comment (willmcb) [inline app/helpers/cases_helper.rb:277]: Small thing but I think we may have a bit of redundant logic here as we are asking if a case is third party twice when we could just have:

```ruby
      if kase.type == "Case::SAR::Offender" && kase.third_party?
        if kase.third_party_company_name.present?
          content_tag(:div, kase.third_party_company_name, class: 'case-name-detail')
        elsif !kase.third_party_company_name.present?
          content_tag(:div, kase.third_party_name, class: 'case-name-detail')
        else
          content_tag(:div, kase.subject, class: 'case-name-detail')
        end
```

Comment (njpreddy) [reply app/helpers/cases_helper.rb]: Thanks Will, I've refactored with this logic.

File: app/helpers/cases_helper.rb [hunk 2]
```diff
@@ -272,10 +272,25 @@ def action_link_for_destroy_case_link(kase, linked_case)
     end
   end
 
+  # def request_details_html(kase)
```
Comment (willmcb) [inline app/helpers/cases_helper.rb:275]: I think we should delete this comment if we aren't using this implementation, could be confusing to someone looking back at the code 👍

**Review by willmcb — APPROVED** [review]

File: app/helpers/cases_helper.rb (hunk 2 above)
Comment (njpreddy) [reply app/helpers/cases_helper.rb]: Commented code now removed.

**Pull request merged**